### PR TITLE
[Enhancement] support the drop CATALOG xx IF EXISTS syntax

### PR DESCRIPTION
**Pull request by zhaohehuhu**
## Why I'm doing:
as title
## What I'm doing:
as title

## What type of PR is this:

- [ ] BugFix
- [ ] Feature
- [x] Enhancement
- [ ] Refactor
- [ ] UT
- [ ] Doc
- [ ] Tool

Does this PR entail a change in behavior?

- [ ] Yes, this PR will result in a change in behavior.
- [x] No, this PR will not result in a change in behavior.

If yes, please specify the type of change:

- [ ] Interface/UI changes: syntax, type conversion, expression evaluation, display information
- [ ] Parameter changes: default values, similar parameters but with different default values
- [ ] Policy changes: use new policy to replace old one, functionality automatically enabled
- [ ] Feature removed
- [ ] Miscellaneous: upgrade & downgrade compatibility, etc.

## Checklist:

- [x] I have added test cases for my bug fix or my new feature
- [ ] This pr needs user documentation (for new or modified features or behaviors)
  - [ ] I have added documentation for my new feature or new function
- [ ] This is a backport pr

## Bugfix cherry-pick branch check:
- [x] I have checked the version labels which the pr will be auto-backported to the target branch
  - [x] 3.3
  - [ ] 3.2
  - [ ] 3.1
  - [ ] 3.0
  - [ ] 2.5

### CODE REVIEW

**Comment by zhaohehuhu** [comment]
@Youngwb @stephen-shelby plz help review it.

File: fe/fe-core/src/main/java/com/starrocks/sql/parser/StarRocks.g4 [hunk 1]
```diff
@@ -715,7 +715,7 @@ showCreateExternalCatalogStatement
     ;
 
 dropExternalCatalogStatement
-    : DROP CATALOG catalogName=identifierOrString
+    : DROP CATALOG catalogName=identifierOrString (IF EXISTS)?
```
Comment (imay) [inline fe/fe-core/src/main/java/com/starrocks/sql/parser/StarRocks.g4:718]:
```suggestion
    : DROP CATALOG (IF EXISTS)? catalogName=identifierOrString
```

Comment (zhaohehuhu) [reply fe/fe-core/src/main/java/com/starrocks/sql/parser/StarRocks.g4:718]: Thanks.

File: fe/fe-core/src/main/java/com/starrocks/common/ErrorCode.java [hunk 2]
```diff
@@ -223,6 +223,8 @@ public enum ErrorCode {
             "Dynamic history partition num must greater than 0"),
     ERR_CATALOG_EXISTED_ERROR(5700, new byte[] {'4', '2', '0', '0', '0'},
             "Catalog '%s' already exists"),
+    ERR_UNKNOWN_CATALOG(5701, new byte[] {'4', '2', '0', '0', '0'},
```
Comment (stephen-shelby) [inline fe/fe-core/src/main/java/com/starrocks/common/ErrorCode.java:226]: `ERR_BAD_CATALOG_ERROR` is enough

File: fe/fe-core/src/main/java/com/starrocks/qe/DDLStmtExecutor.java [hunk 3]
```diff
@@ -934,6 +934,15 @@ public ShowResultSet visitCreateCatalogStatement(CreateCatalogStmt stmt, Connect
         @Override
         public ShowResultSet visitDropCatalogStatement(DropCatalogStmt stmt, ConnectContext context) {
             ErrorReport.wrapWithRuntimeException(() -> {
+                String catalogName = stmt.getName();
+                if (!context.getGlobalStateMgr().getCatalogMgr().catalogExists(catalogName)) {
+                    if (stmt.isIfExists()) {
+                        LOG.info("drop catalog[{}] which does not exist", catalogName);
```
Comment (stephen-shelby) [inline fe/fe-core/src/main/java/com/starrocks/qe/DDLStmtExecutor.java:940]: Failed to drop catalog xxxxx

File: fe/fe-core/src/main/java/com/starrocks/common/ErrorCode.java (hunk 2 above)
Comment (zhaohehuhu) [reply fe/fe-core/src/main/java/com/starrocks/common/ErrorCode.java]: Done. Thanks.

File: fe/fe-core/src/main/java/com/starrocks/qe/DDLStmtExecutor.java (hunk 3 above)
Comment (zhaohehuhu) [reply fe/fe-core/src/main/java/com/starrocks/qe/DDLStmtExecutor.java:940]: I prefer to keep this message to make it consistent with above message style, eg. "drop storage volume[{}] which does not exist"

**Review by imay — APPROVED** [review]

**Review by stephen-shelby — APPROVED** [review]

**Review by packy92 — APPROVED** [review]

**Review by wangsimo0 — APPROVED** [review]

**Pull request merged**